### PR TITLE
conversion to UDv2 guidelines using ud.Convert1to2 ud.he.FixNeg

### PR DESCRIPTION
**Pull request by martinpopel**
@yoavg @rtsarfaty: Are you working on the UDv2 update of UD_Hebrew?
Do you know today is the data freeze deadline and http://universaldependencies.org/validation.html shows FAIL?
It would be pity to miss Hebrew in the UDv2 release.

I don't know much about Hebrew, but this commit makes the data valid at least.
See https://github.com/udapi/udapi-python/tree/master/udapi/block/ud

Do you have the original texts?
If not, I will try to adapt [setspaceafter.py](https://github.com/udapi/udapi-python/blob/master/udapi/block/ud/setspaceafter.py)
so it works also with [Hebrew-specific punctuation](https://en.wikipedia.org/wiki/Hebrew_punctuation).

### CODE REVIEW

**Comment by rtsarfaty** [comment]
yes, we are definitely working on it. we have a treebank in progress offline, we havent pushed any update yet. we will be busy almost until the deadline fixing errors reported by the svalidation, and some other hebrew specific improvents. would it be fair to assume that 15 of feb ends at midnight gmt-12?

**Comment by martinpopel** [comment]
> yes, we are definitely working on it

Great. I hoped so.

> fixing errors reported by the svalidation

Just out of curiosity: have you installed [it](http://universaldependencies.org/svalidation.html) offline?
Here are the svalidation bugs left after my conversion in this PR:
http://ufallab.ms.mff.cuni.cz/~popel/he/bugs.html
As you can see [here](https://github.com/UniversalDependencies/UD_Hebrew/blob/4c3a6dc29b3f26bc0c8088f185a026d0b52a838c/not-to-release/convert.sh#L10) I have omitted `no-(VerbForm|NumType|PronType)|det-upos` as there were too many of those errors.
If you want I can generate such html for your newest version (once you push it to the dev branch or send me by email).

BTW: are you sure `HebSource=ConvUncertainHead` and `HebSource=ConvUncertainLabel` should stay in *morphological* FEATS? I think these belong to MISC.

> would it be fair to assume that 15 of feb ends at midnight gmt-12?

I hope so. There are many more treebanks still failing to pass `validate.py` which I can fix with my scripts in Udapi.
Let me know if you need any help with automatic transformations.
See e.g. [ud.he.FixNeg](https://github.com/udapi/udapi-python/blob/master/udapi/block/ud/he/fixneg.py) for an example how to write transformation rules in Udapi/Python.

**Comment by yoavg** [comment]
Hi all,
As @rtsarfaty said, we are working on it.

@martinpopel , I assume Hebrew is not the only treebank you are attempting to automatically validate using your Udapi scripts and contacting the owners -- I think this may not be the prefered way to resolve this. For example, many of our failure cases are "Copula is not AUX". I can imagine that a way to resolve them would be to change the POS-tag to AUX -- but this will hide the underlying issue which is "this construction is in some likelihood not a copular one". 
In other words, the failing validation surfaces problems in the underlying annotation, and I don't think that hiding these by brute-force fixing of the validation is a good idea. Let these things fail, and let the owners look at the problems on a case-by-case basis and decide what to do.

yoav

**Comment by dan-zeman** [comment]
> would it be fair to assume that 15 of feb ends at midnight gmt-12?

Yes.

**Comment by martinpopel** [comment]
@yoavg: Thanks for your opinion. I am happy you will take care of UD_Hebrew and I can close this PR.

Yes, I am trying to help fixing several treebanks. As I wrote, I think it would be pity to loose some treebanks from the UDv2 release (and CoNLL2017).
I always try to contact the treebank owners/maintainers, so they can check and approve all the changes I suggest.

I have several kinds of scripts:
* [ud.Convert1to2.py](https://github.com/udapi/udapi-python/blob/master/udapi/block/ud/convert1to2.py) converts only the very clear cases, the rest is marked with `ToDo` in MISC to be resolved manually. Similarly [ud.SetSpaceAfterFromText](https://github.com/udapi/udapi-python/blob/master/udapi/block/ud/setspaceafterfromtext.py) fills `SpaceAfter=No` according to the `text`, but skips (and marks) sentences with unexpected inconsistencies.
* [ud.SetSpaceAfter](https://github.com/udapi/udapi-python/blob/master/udapi/block/ud/setspaceafter.py) reconstructs `SpaceAfter=No` and `text` heuristically, but it should be used only as the last resort, if the original texts were lost. I try to inspect the results manually.
* There are several treebank-specific scripts - focusing on a given transformation (e.g. splitting underscored tokens into flat structures), again the result should be manually inspected.
* [ud.MarkBugs](https://github.com/udapi/udapi-python/blob/master/udapi/block/ud/markbugs.py) is an improved version of svalidation.html, it does not change anything, just marks bugs.

I hope I have explained my goal is not to brute-force fix/hide any errors. Of course, there may be errors left in my scripts and in the data, but manual editing may result in errors as well.

> our failure cases are "Copula is not AUX"

In UDv1.4 it was common to tag copulas as VERB. We should distinguish two parts of UDv2 innovations: First, there are much more detailed guidelines which constructions should be considered copula and which not. This I cannot check/fix automatically (and I mention it in ud.Convert1to2 documentation). Second, all deprel=cop *must* have upos=AUX or PRON, which is easy to implement.

These two issues should not be confused with the אינם VerbType=Cop occurrences I noticed in UD_Hebrew v1.4. Here I wrote a special script [ud.he.FixNeg](https://github.com/udapi/udapi-python/blob/master/udapi/block/ud/he/fixneg.py) which changes these to AUX/cop/Polarity=Neg. This was meant rather as a suggestion (as I don't know much about Hebrew). In general the whole PR was just a simple way how to show you what are the possibilities. I did not expect you to merge it. I am sorry I have not explained my goals properly before doing the PR, but as you know the deadline is very close and there was no activity in the UD_Hebrew dev branch.

**Comment by yoavg** [comment]
Thanks @martinpopel , now things are much clearer, and I have no objections whatsoever! so keep up this very good work you are doing! :)

**Comment by martinpopel** [comment]
Thanks @yoavg for your response.
BTW: here are the svalidation errors after PR #12 : http://ufallab.ms.mff.cuni.cz/~popel/he/bugs-PR12.html (excluding `no-(VerbForm|NumType|PronType)`).

Most/all `appos-chain` are false alarms I guess, i.e. they are really nested appositions, which are OK. I will try to fix the test.

In addition to [svalidation.html](http://universaldependencies.org/svalidation.html) `ud.MarkBugs` shows also `goeswith-gap` and `case-child` (these may be false alarms as well).
